src/Database/Query/SelectQuery.php

```diff
@@ -232,6 +234,40 @@ public function distinct(ExpressionInterface|array|string|bool $on = [], bool $o
         return $this;
     }
 
+    /**
+     * Add engine-specific optimizer hint.
+     *
+     * @param string $name Hint name
+     * @param array|string $parameters Hint parameters
+     * @param bool $overwrite Whether to replace existing hints
+     * @return $this
+     */
+    public function hint(string $name, array|string $parameters, bool $overwrite = false)
+    {
+        $hints[] = compact('name', 'parameters');
+        $this->_parts['hint'] = $overwrite ? $hints : array_merge($this->_parts['hint'], $hints);
+        $this->_dirty();
+
+        return $this;
+    }
+
+    /**
+     * Add mysql-specific index hint.
+     *
+     * @param string $type Hint type
+     * @param array|string $names Index names
+     * @param bool $overwrite Whether to replace existing hints
+     * @return $this
+     */
+    public function index(string $type, array|string $names, bool $overwrite = false)
```

Comment:
Could this `type` parameter be narrowed with an enum? Or could we validate it with a list of valid names?

We could even provide a list of valid options, or a validation enum via the schema dialect. SchemaDialects are a good source for driver specific options like this. If the list of options is `null` we could return early 🤷 That could let us have both a validated string/constant/enum based API and have it be implemented optionally at the same time.